### PR TITLE
[DX][Core][Shop] Simplified variant price rendering

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets | - |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Templating/Helper/ChannelBasedPriceHelper.php [hunk 1]
```diff
@@ -12,49 +12,38 @@
...
-use Sylius\Component\Core\Model\OrderInterface;
 use Sylius\Component\Core\Model\ProductVariantInterface;
-use Sylius\Component\Order\Context\CartContextInterface;
 use Symfony\Component\Templating\Helper\Helper;
+use Webmozart\Assert\Assert;
 
 /**
  * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
  */
 class ChannelBasedPriceHelper extends Helper implements ChannelBasedPriceHelperInterface
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Templating/Helper/ChannelBasedPriceHelper.php]: Rename to `PriceHelper`

File: src/Sylius/Bundle/CoreBundle/Templating/Helper/ChannelBasedPriceHelper.php [hunk 2]
```diff
@@ -12,49 +12,38 @@
...
+    public function __construct(ProductVariantPriceCalculatorInterface $productVariantPriceCalculator)
+    {
         $this->productVariantPriceCalculator = $productVariantPriceCalculator;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getPriceForCurrentChannel(ProductVariantInterface $productVariant)
+    public function getPriceForChannel(ProductVariantInterface $productVariant, array $context)
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Templating/Helper/ChannelBasedPriceHelper.php]: Simply `getPrice`, the channel is not required.

File: src/Sylius/Bundle/CoreBundle/Templating/Helper/ChannelBasedPriceHelperInterface.php [hunk 3]
```diff
@@ -20,8 +20,11 @@
 {
     /**
      * @param ProductVariantInterface $productVariant
+     * @param array $context
      *
      * @return int
+     *
+     * @throws \InvalidArgumentException if $context doesn't have a 'channel' key
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Templating/Helper/ChannelBasedPriceHelperInterface.php]: Skip the `throws`.

File: src/Sylius/Bundle/CoreBundle/Twig/ChannelBasedPriceExtension.php [hunk 4]
```diff
@@ -37,7 +37,7 @@ public function __construct(ChannelBasedPriceHelperInterface $helper)
     public function getFilters()
     {
         return [
-            new \Twig_SimpleFilter('sylius_channel_variant_price', [$this->helper, 'getPriceForCurrentChannel']),
+            new \Twig_SimpleFilter('sylius_channel_variant_price', [$this->helper, 'getPriceForChannel']),
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Twig/ChannelBasedPriceExtension.php]: `sylius_channel_variant_price` -> `sylius_calcualate_price`

**Comment by pjedrzejewski** [comment]
Great work Jasiek, thanks! :)

**Pull request merged**